core/database_api.php

```diff
@@ -1263,3 +1263,54 @@ function db_format_query_log_msg( $p_query, array $p_arr_parms ) {
 	}
 	return $p_query;
 }
+
+/**
+ * Returns true if a specific capability is suported in the current database server,
+ * false otherwise.
+ *
+ * @param integer $p_capability   See DB_CAPABILITY_* constants
+ * @return boolean    True if the capability is supported, false otherwise.
+ */
+function db_has_capability( $p_capability ) {
+	static $s_cache = array();
+	if( !isset( $s_cache[$p_capability] ) ) {
+		$s_cache[$p_capability] = db_test_capability( $p_capability );
+	}
+	return $s_cache[$p_capability];
+}
+
+/**
+ * Tests if a specific capability is suported in the current database server.
+ *
+ * @param integer $p_capability   See DB_CAPABILITY_* constants
+ * @return boolean    True if the capability is supported, false otherwise.
+ */
+function db_test_capability( $p_capability ) {
+	global $g_db, $g_db_functional_type;
+	$t_server_info = $g_db->ServerInfo();
+
+	switch( $p_capability ) {
+		case DB_CAPABILITY_WINDOW_FUNCTIONS:
+			switch( $g_db_functional_type ) {
+				case DB_TYPE_ORACLE: # since 8i
+				case DB_TYPE_PGSQL: # since 8.4
+				case DB_TYPE_MSSQL: # since 2008
+					return true;
+				case DB_TYPE_MYSQL:
+					# mysql, since 8.0.2
+					if( version_compare( $t_server_info['version'], '8.0.2', '>=' )
+							&& false !== stripos( $t_server_info['description'], 'mysql' ) ) {
+						return true;
+					}
+					# mariaDB, since 10.2
+					if( version_compare( $t_server_info['version'], '10.2', '>=' )
```

Comment:
What about if MariaDB version is >= 8.0.2 and < 10.2 ?
Maybe we need to identify if we are on MariaDB or MySQL ?

Reply:
> Maybe we need to identify if we are on MariaDB or MySQL

looking in the description for appearance of "mysql" or mariadb" strings.

Reply:
if the string does not appear in the descriptions (seems like some instances do so), failsafe to not supported...